server/datastore/mysql/setup_experience.go

```diff
@@ -2,15 +2,122 @@ package mysql
 
 import (
 	"context"
+	"database/sql"
+	"fmt"
 
 	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
 	"github.com/fleetdm/fleet/v4/server/fleet"
 	"github.com/jmoiron/sqlx"
 )
 
+func (ds *Datastore) GetSetupExperienceScript(ctx context.Context, teamID *uint) (*fleet.Script, error) {
+	query := `
+SELECT
+  id,
+  team_id,
+  name,
+  script_content_id,
+  created_at,
+  updated_at
+FROM
+  setup_experience_scripts
+WHERE
+  global_or_team_id = ?
+`
+	var globalOrTeamID uint
+	if teamID != nil {
+		globalOrTeamID = *teamID
+	}
+
+	var script fleet.Script
+	if err := sqlx.GetContext(ctx, ds.reader(ctx), &script, query, globalOrTeamID); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, ctxerr.Wrap(ctx, notFound("SetupExperienceScript"), "get setup experience script")
+		}
+		return nil, ctxerr.Wrap(ctx, err, "get setup experience script")
+	}
+
+	return &script, nil
+}
+
+func (ds *Datastore) SetSetupExperienceScript(ctx context.Context, script *fleet.Script) error {
+	// var res sql.Result
+	err := ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		var err error
+
+		// first insert script contents
+		scRes, err := insertScriptContents(ctx, tx, script.ScriptContents)
+		if err != nil {
+			return err
+		}
+		id, _ := scRes.LastInsertId()
+
+		// then create the script entity
+		_, err = insertSetupExperienceScript(ctx, tx, script, uint(id))
+		return err
+	})
+
+	// // TODO: Do we want to return the script here?
```

Comment:
I'd lean towards returning it, seems like it always comes in handy (at least in tests). Not a strong preference tho.